docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/index.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: docs
+title: Creating Custom Reporter
+permalink: /documentation/extending-testcafe/custom-reporter-plugin/creating-reporter-plugin/
+---
+# Creating Reporter Plugin
+
+## Generating Reporter Project
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-reporter` using [npm](https://www.npmjs.com/) (we assume you have pre-installed [node.js](https://nodejs.org/)).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-reporter
+```
+
+Create a new directory where the generator should place your scaffolded project files and go into it.
+
+```bash
+mkdir MyReporter
+cd MyReporter
+```
+
+Then run the reporter generator to create a new project:
+
+```bash
+yo testcafe-reporter
+```
+
+The generator will provide optional settings to customize your reporter. You need to specify:
+
+* a reporter name. The generator will suggest the reporter name that matches the reporter directory name. You can keep this name or specify another one.
+* your GitHub username.
+* the URL of your website.
+* whether you want to implement a custom error decorator for the reporter.
+
+Yeoman will automatically scaffold out your reporter, install the required dependencies, and pull in several useful Gulp tasks for your workflow.
+
+## Implementing Reporter
+
+Since the reporter has been scaffolded out, go to the reporter directory and open the `src/index.js` file.
+
+You have to implement four methods which will be further mixed in into the reporter.
+
+```js
+reportTaskStart (/* startTime, userAgents, testCount */) {
+    throw new Error('Not implemented');
+},
+
+reportFixtureStart (/* name, path */) {
+    throw new Error('Not implemented');
+},
+
+reportTestDone (/* name, errs, durationMs, unstable, screenshotPath */) {
+    throw new Error('Not implemented');
+},
+
+reportTaskDone (/* endTime, passed */) {
+    throw new Error('Not implemented');
+}
+```
+
+These methods should output the desired information at certain moments during test run.
+
+* **reportTaskStart** - fires when a test task is started, that is at the beginning of testing.
+
+* **reportFixtureStart** - fires each time a fixture is started.
+
+* **reportTestDone** - fires each time a test is finished.
+
+* **reportTaskDone** - fires when the entire task is finished, that is at the end of testing.
+
+All the required data is provided to these methods through their parameters.
+
+To output information, use [helper methods](/testcafe/documentation/extending-testcafe/custom-reporter-plugin/helper-methods/) and libraries exposed by the object. To refer to the helper methods, use `this`.
```

Comment:
It's not quite clear. You need to say that TestCafe will mixin useful helper methods to your reporter, which you can access using `this` keyword.